feat: set Swift active compilation conditions build setting to DEBUG

okwasniewski opened the pull request:
## Summary:

This PR adds cocoapods utility to set `SWIFT_ACTIVE_COMPILATION_CONDITIONS` to DEBUG, which is set to this value by default (when generating a new native Xcode project). 

This allows to use the `#if DEBUG` compilator directive in Swift to work out of the box, without any changes on user's side: 

```swift
override func bundleURL() -> URL? {
#if DEBUG
    RCTBundleURLProvider.sharedSettings().jsBundleURL(forBundleRoot: "index")
#else
    Bundle.main.url(forResource: "main", withExtension: "jsbundle")
#endif
  }
``` 

## Changelog:

[IOS] [ADDED] - Set SWIFT_ACTIVE_COMPILATION_CONDITIONS to DEBUG 

## Test Plan:

Run `bundle exec pod install` and check if the active compilation flags are populated: 

![CleanShot 2024-01-17 at 13 11 03@2x](https://github.com/facebook/react-native/assets/52801365/68f119a6-af47-41bc-a1f1-9085fe4df6a0)

okwasniewski commented on packages/react-native/scripts/cocoapods/utils.rb:
```diff
@@ -73,6 +73,20 @@ def self.set_use_hermes_build_setting(installer, hermes_enabled)
         end
     end
 
+    def self.set_swift_compilation_conditions(installer)
```
Maybe we could create a utility to set any flag by name and value because for any flags in the future this code will be redundant

okwasniewski commented:
Hey @cipolleschi, anything I can do to help you get this one merged?

cipolleschi reviewed (changes requested):
Sorry, I completely missed this PR. 😭 

The changes look good. Could you add a test in `utils-test.rb` file to verify that the build settings is actually added? In this way we can make sure not to miss changes in the `set_swift_compilation_conditions` method

okwasniewski commented:
@cipolleschi I've refactored the method to `set_build_setting` which accepts necessary arguments to set proper build settings. This method can remove `set_use_hermes_build_setting` and `set_node_modules_user_settings` methods (less code to maintain). I've also changed the tests to only test the utility

cipolleschi reviewed (approved):
Great job, thanks for refactoring!

cipolleschi commented:
/rebase - _this command rebase the pr on top of main automatically_

okwasniewski commented:
Hey @cipolleschi, is there anything more we need to get done in order to merge this PR?

cipolleschi commented:
/rebase - this comment automatically rebase on top of main
